uaclient/entitlements/base.py

```diff
@@ -296,6 +296,7 @@ def __init__(
             self.extra_args = []
         self._called_name = called_name
         self._valid_service = None  # type: Optional[bool]
+        self._is_main_repo_updated = False
```

Comment:
`main` has a different meaning when speaking on repositories, and updating the whole sources.list will give you more than `main`. A nit, I know, but I'd call this something like `_is_sources_list_updated`  to better reflect what it means.

Reply:
Updated.